src/lib/state.ts

```diff
@@ -288,3 +290,23 @@ export const dumpState = (
     ),
   };
 };
+
+export const getTransformedValueMetadata = (
+  node: ZWaveNode,
+  valueArgs: ValueID,
+  schemaVersion: number
+): ValueMetadata => {
+  let metadata: ValueMetadata;
+  metadata = node.getValueMetadata(valueArgs);
+  schemaTransformValueMetadata(metadata, schemaVersion);
+  return metadata;
+};
+
+export const schemaTransformValueMetadata = (
+  metadata: ValueMetadata,
+  schemaVersion: number
+): void => {
+  if (schemaVersion < 2 && metadata.type === "buffer") {
+    metadata.type = "string";
```

Comment:
Here you're actually changing a ZJS type. When we copied args to newArgs, we did a shallow clone, only the top level object has been copied. All values are still pointing at the same instance.

I think that instead of trying to pass classes to JSON serialize, we should instead write out the types that we want like we do for node.

Reply:
Not totally following the second part. Are you suggesting we create a `MetadataState` like we have a `NodeState` and `ValueState` and add a `dumpMetadata` function that explicitly copies the properties over while updating `metadata.type` if needed?

Reply:
Not sure if I took the right path here but attempted to follow your suggestion in the latest commit.